website/en/docs/types/functions.md

```diff
@@ -213,6 +213,54 @@ var num: number = method.call(42);
 var str: string = method.call(42);
 ```
 
+### Functions with %checks <a class="toc" id="toc-function-checks" href="#toc-function-checks"></a>
+
+Sometimes you will want to move condition from `if` statement into a function
+
+```js
+function concat(a: ?string, b: ?string): string {
+  if (a && b) {
+    return a + b;
+  }
+  return '';
+}
+```
+
+However, `flow` will see an error in the code below:
+
+```js
+function truthy(a, b): boolean {
+  return a && b;
+}
+
+function concat(a: ?string, b: ?string): string {
+  if (truthy(a, b)) {
+    // $ExpectError
+    return a + b;
+  }
+  return '';
+}
+```
+
+You can fix it using `%checks` mark like this:
+
+```js
+function truthy(a, b): boolean %checks {
+  return !!a && !!b;
+}
+
+function concat(a: ?string, b: ?string): string {
+  if (truthy(a, b)) {
+    return a + b;
+  }
+  return '';
+}
+```
+
+Such predicate functions must return immediately (i.e. you can not declare variables). 
+Also, it's not possible to call other predicate functions inside a predicate function. 
```

Comment:
This is possible too. See https://github.com/facebook/flow/blob/master/tests/predicates-inferred/simple-predicate-func.js#L38.